lib/Mojo/WebSocketProxy/Backend/ConsumerGroups.pm

```diff
@@ -309,7 +311,7 @@ sub _on_message {
 
     my $message = eval { decode_json_utf8($raw_message) };
 
-    unless(ref $message eq 'HASH' && $message->{message_id}) {
+    if(ref $message ne ref {} && !$message->{message_id}) {
```

Comment:
> `ref $message ne ref {}`

Please don't ever do this, it's only going to cause confusion...

We should have a separate `try` block for the JSON decoding, and return with an error on failure, then have a *separate* check for `message_id`.

Reply:
Changed and fixed, thanks